sdk/inc/azure/core/_az_cfg.h

```diff
@@ -25,8 +25,10 @@
 // automatic variable '...'
 #pragma warning(disable : 4221)
 
-// warning C6011: Dereferencing NULL pointer. Using _az_PRECONDITION_NOT_NULL
-#pragma warning(disable : 6011)
+// warning C28278 : Function appears with no prototype in scope. Only limited analysis can be
+// performed. Include the appropriate header or add a prototype. This warning also occurs if
+// parameter or return types are omitted in a function definition.
+#pragma warning(disable : 28278)
```

Comment:
I think this is a good warning to keep enabled, and opt-out in the specific instances. If it is too noisy, then ignoring it is fine. Up to you.

Reply:
My opinion is that the tool should be smart enough to figure out the signature itself. I did the minimum effort to declare the function before defining it, but that wasnt enough. I suspect if I not only declare it, but also remove static, it would make the warning to go away, but it is asking too much. This isn't about the problem with our code, it is the code analysis tool tells us that its current version is not smart enough to figure out all the info, and asing for help. I could put a couple of extra line, but I don't think we should change the design to compensate for code analysis current weaknesses.